rust/metaplex/program/src/processor/set_store.rs

```diff
@@ -88,3 +86,108 @@ pub fn process_set_store<'a>(
     store.serialize(&mut *store_info.data.borrow_mut())?;
     Ok(())
 }
+
+pub fn process_set_store<'a>(
+    program_id: &'a Pubkey,
+    accounts: &'a [AccountInfo<'a>],
+    public: bool,
+) -> ProgramResult {
+    let account_info_iter = &mut accounts.iter();
+
+    let store_info = next_account_info(account_info_iter)?;
+    let admin_wallet_info = next_account_info(account_info_iter)?;
+    let payer_info = next_account_info(account_info_iter)?;
+    let token_program_info = next_account_info(account_info_iter)?;
+    let token_vault_program_info = next_account_info(account_info_iter)?;
+    let token_metadata_program_info = next_account_info(account_info_iter)?;
+    let auction_program_info = next_account_info(account_info_iter)?;
+    let system_info = next_account_info(account_info_iter)?;
+    let rent_info = next_account_info(account_info_iter)?;
+
+    set_store_logic(
+        public,
+        program_id,
+        auction_program_info,
+        token_vault_program_info,
+        rent_info,
+        system_info,
+        token_metadata_program_info,
+        token_program_info,
+        store_info,
+        admin_wallet_info,
+        payer_info,
+    )
+}
+
+pub fn process_set_store_v2<'a>(
+    program_id: &'a Pubkey,
+    accounts: &'a [AccountInfo<'a>],
+    public: bool,
+    settings_uri: Option<String>,
+) -> ProgramResult {
+    let account_info_iter = &mut accounts.iter();
+
+    let store_info = next_account_info(account_info_iter)?;
+    let store_config_info = next_account_info(account_info_iter)?;
+    let admin_wallet_info = next_account_info(account_info_iter)?;
+    let payer_info = next_account_info(account_info_iter)?;
+    let token_program_info = next_account_info(account_info_iter)?;
+    let token_vault_program_info = next_account_info(account_info_iter)?;
+    let token_metadata_program_info = next_account_info(account_info_iter)?;
+    let auction_program_info = next_account_info(account_info_iter)?;
+    let system_info = next_account_info(account_info_iter)?;
+    let rent_info = next_account_info(account_info_iter)?;
+
+    let res = set_store_logic(
+        public,
+        program_id,
+        auction_program_info,
+        token_vault_program_info,
+        rent_info,
+        system_info,
+        token_metadata_program_info,
+        token_program_info,
+        store_info,
+        admin_wallet_info,
+        payer_info,
+    );
```

Comment:
it would be better to use " ? " operator here so we don't need check result with " if "